msm/src/circuit_design/composition.rs

```diff
@@ -235,6 +265,10 @@ impl<
         self.0.assert_zero(cst);
     }
 
+    fn set_assert_mapper(&mut self, mapper: Box<dyn Fn(Self::Variable) -> Self::Variable>) {
+        self.0.set_assert_mapper(mapper);
```

Comment:
I am not sure I like the `self.0` notation. Is it a tuple? If so, wouldn't it make sense to make it a struct instead?

Reply:
That's the standard way to do wrappers in rust: `struct Wrapper(pub MyType)` and then `let wrapper = Wrapper(mytype); let y = wrapper.0;`. Again, discussable, but perhaps not in this PR?..

Reply:
Sure sure, not a blocking comment